xinference/model/llm/pytorch/core.py

```diff
@@ -142,9 +143,25 @@ def load(self):
             "trust_remote_code"
         )
         model_format = self.model_spec.model_format
+
+        is_device_map_auto = False
+
+        # This is required for Intel GPU to actually work with accelerate device_map until
+        # https://github.com/intel/intel-extension-for-pytorch/issues/522
+        # is resolved
+        max_memory_env = os.getenv("ACCELERATE_MAX_MEMORY", None)
+
+        if max_memory_env is not None:
+            max_memory_raw = json.loads(max_memory_env)
+            max_memory = {
+                int(k) if k.isdigit() else k: max_memory_raw[k] for k in max_memory_raw
+            }
+            kwargs["max_memory"] = max_memory
```

Comment:
Better to add a class method like `config_xpu`, and only call it when device is xpu.

Reply:
For now, due to https://github.com/intel/intel-extension-for-pytorch/issues/522, Intel GPU requires it to use cpu offloading, but this code can be used for other devices too, if someone wants to customize the memory map for `accelerate`. Or we could expose this to some global config or in `PytorchModel._pytorch_model_config`?

Reply:
It's OK if it also works for other devices.